.github/workflows/3rd-party-tests.yml

```diff
@@ -0,0 +1,110 @@
+name: Test on 3rd party libraries
+
+on:
+  push:
+    branches:
+      - "master"
+      - "sqlalchemy_pipeline"
+    paths-ignore:
+      - "docs/*"
+      - "tools/*"
+  workflow_dispatch:
+
+concurrency:
+  # Cancel older requests of the same workflow in the same branch.
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  sqlalchemy:
+    name: sqlalchemy-${{ matrix.sqlalchemy }},psycopg-${{ matrix.impl }},py-${{ matrix.python-version }}
+    # linux should be enough to test if everything works.
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version:
+          - "3.11"
+          - "3.7"
+        sqlalchemy:
+          - git_main
+          - release
+        impl:
+          - c
+          - python
+
+    env:
+      PSYCOPG_IMPL: ${{ matrix.impl }}
+      DEPS: ./psycopg[test] ./psycopg_pool pytest pytest-xdist
+
+    services:
+      pg:
+        image: postgres:15
+        env:
+          POSTGRES_PASSWORD: password
+          POSTGRES_DB: test
+        ports:
+          - 5432:5432
+
+    steps:
+      - name: Select sqlalchemy pip target
+        run: |
+          case ${{ matrix.sqlalchemy }} in
+            git_main)
+              echo "PIP_SQLALCHEMY=git+https://github.com/sqlalchemy/sqlalchemy.git#egg=SQLAlchemy" >> $GITHUB_ENV
+              ;;
+            release)
+              echo "PIP_SQLALCHEMY=sqlalchemy>=2a" >> $GITHUB_ENV
+              ;;
+            *)
+              echo "Unknown value for sqlalchemy: ${{ matrix.sqlalchemy }}"
+              exit 1
+              ;;
+          esac
```

Comment:
I would prefer to have these entries directly in the matrix and used later at `pip download`, without this intermediate step.

Reply:
I did this for the name of the job. I'm not sure if I can otherwise create something meaningful. Maybe GitHub supports json objects as variable values? I'll do somr tries

Reply:
I may have found a reasonable solution to have both a label and the variable defined in the matrix section